google_maps/public/js/src/google_maps_edit.js

```diff
@@ -76,8 +80,42 @@ function GoogleMapsEditXBlock(runtime, element) {
   $(element).on('click', '.clear-iframe-height', function() {
     setDefaultValue(iframeHeightInput, iframeHeightDefaultValue, iframeHeightClearButton);
   });
-  
+
   $(element).on('click', '.save-button', saveXBlockData);
 
   $(element).on('click', '.cancel-button', exitEditMode);
+
 }
+
+googleMapsXBlockApp.initMap = function () {
+  var input = document.getElementById('place-name-input'),
+      autocomplete = new google.maps.places.Autocomplete(input),
+      placeNameInput = $('#place-name-input'),
+      preparedParameters = '',
+      getParameters = {};
+
+  autocomplete.addListener('place_changed', function () {
+    var place = autocomplete.getPlace();
+
+    if (!place.geometry) {
+      // User entered the name of a Place that was not suggested and
+      // pressed the Enter key, or the Place Details request failed.
+      $('.action-item .save-button').addClass('is-disabled');
+      placeNameInput.addClass('invalid-input').siblings('.error-message').removeClass('hidden');
+      return;
+    } else {
+      $('.action-item .save-button').removeClass('is-disabled');
+      placeNameInput.removeClass('invalid-input').siblings('.error-message').addClass('hidden');
+      googleMapsXBlockApp.placeId = place.place_id;
+    }
+
+    getParameters = {
+      key: googleMapsXBlockApp.googleMapsApiKey,
+      q: 'place_id:' + place.place_id,
+      zoom: 12
+    };
+    preparedParameters = $.param(getParameters);
```

Comment:
You don't need `getParameters` variable since you're using it only in one place.